### PR TITLE
Case 20240: Better material entity targeting

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/20240/Better-material-targeting

Expands the functionality of the `parentMaterialName` property like this:
```
@property {string} parentMaterialName="0" - Selects the mesh part or parts within the parent to which to apply the material.

- If  in the format `"mat::string"`, all mesh parts with material name `"string"` are replaced.
- Otherwise the property value is parsed as an unsigned integer, specifying the mesh part index to modify.  If `"all"`,
- all mesh parts will be replaced.  If an array (starts with `"["` and ends with `"]"`), the string will be
- split at each `","` and each element will be parsed as either a number or a string if it starts with
- `"mat::"`.  In other words, `"[0,1,mat::string,mat::string2]"` will replace mesh parts 0 and 1, and any
- mesh parts with material `"string"` or `"string2"`.  Do not put spaces around the commas.  Invalid values
- are parsed to `0`.
```

Test plan:
- run the content/entity/material tests from this branch: https://github.com/highfidelity/hifi_tests/tree/b0a6a37e47f3a19105dddc172823bf21de5e6f79/tests/content/entity/material

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16072/HighFidelity-Beta-PR14619-3578df7.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16072/HighFidelity-Beta-PR14619-3578df7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16072/HighFidelity-Beta-Interface-PR14619-3578df7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16072/HighFidelity-Beta-PR14619-3578df7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16072/HighFidelity-Beta-Interface-PR14619-3578df7.exe

File: libraries/entities/src/EntityItemProperties.cpp [hunk 1]
```diff
@@ -779,10 +779,14 @@ EntityPropertyFlags EntityItemProperties::getChangedProperties() const {
  * @property {number} priority=0 - The priority for applying the material to its parent. Only the highest priority material is 
  *     applied, with materials of the same priority randomly assigned. Materials that come with the model have a priority of 
  *     <code>0</code>.
- * @property {string|number} parentMaterialName="0" - Selects the submesh or submeshes within the parent to apply the material 
- *     to. If in the format <code>"mat::string"</code>, all submeshes with material name <code>"string"</code> are replaced. 
- *     Otherwise the property value is parsed as an unsigned integer, specifying the mesh index to modify. Invalid values are 
- *     parsed to <code>0</code>.
+ * @property {string} parentMaterialName="0" - Selects the mesh part or parts within the parent to which to apply the material.
+ *     If in the format <code>"mat::string"</code>, all mesh parts with material name <code>"string"</code> are replaced.
+ *     Otherwise the property value is parsed as an unsigned integer, specifying the mesh parth index to modify.  If <code>"all"</code>,
```
Comment (ZappoMan) [inline libraries/entities/src/EntityItemProperties.cpp]: parth? do you mean part?

Comment (SamGondelman) [reply libraries/entities/src/EntityItemProperties.cpp]: *parth*

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16558/HighFidelity-Beta-PR14619-17ac413.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16558/HighFidelity-Beta-PR14619-17ac413.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16558/HighFidelity-Beta-Interface-PR14619-17ac413.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16662/HighFidelity-Beta-PR14619-17ac413.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16662/HighFidelity-Beta-PR14619-17ac413.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16662/HighFidelity-Beta-Interface-PR14619-17ac413.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16662/HighFidelity-Beta-PR14619-17ac413.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16662/HighFidelity-Beta-Interface-PR14619-17ac413.exe

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by samcake** [comment]
Good to go

**Pull request merged**